core/chains/evm/gas/fee_history_estimator.go

```diff
@@ -0,0 +1,424 @@
+				nonZeroRewardsLen += 1
+			}
+			// We take the max value for the bumping threshold
+			if reward[1].Cmp(big.NewInt(0)) > 0 {
+				priorityFeeThreshold = bigmath.Max(priorityFeeThreshold, reward[1])
+			}
+		}
+
+		if nonZeroRewardsLen == 0 || priorityFeeThreshold.Cmp(big.NewInt(0)) == 0 {
+			return nil
+		}
+		priorityFeeThresholdWei = assets.NewWei(priorityFeeThreshold)
+		maxPriorityFeePerGas = assets.NewWei(priorityFee.Div(priorityFee, big.NewInt(nonZeroRewardsLen)))
+	}
+	// baseFeeBufferPercentage is added on top as a safety to catch fluctuations in the next blocks.
+	maxFeePerGas := nextBaseFee.AddPercentage(BaseFeeBufferPercentage).Add(maxPriorityFeePerGas)
+
+	promFeeHistoryEstimatorBaseFee.WithLabelValues(f.chainID.String()).Set(float64(nextBaseFee.Int64()))
+	promFeeHistoryEstimatorMaxPriorityFeePerGas.WithLabelValues(f.chainID.String()).Set(float64(maxPriorityFeePerGas.Int64()))
+	promFeeHistoryEstimatorMaxFeePerGas.WithLabelValues(f.chainID.String()).Set(float64(maxFeePerGas.Int64()))
+
+	f.logger.Debugf("Fetched new dynamic prices, nextBlock#: %v - oldestBlock#: %v - maxFeePerGas: %v - maxPriorityFeePerGas: %v - maxPriorityFeeThreshold: %v",
+		nextBlock, feeHistory.OldestBlock, maxFeePerGas, maxPriorityFeePerGas, priorityFeeThresholdWei)
+
+	f.priorityFeeThresholdMu.Lock()
+	f.priorityFeeThreshold = priorityFeeThresholdWei
+	f.priorityFeeThresholdMu.Unlock()
+
+	f.dynamicPriceMu.Lock()
+	defer f.dynamicPriceMu.Unlock()
+	f.dynamicPrice.FeeCap = maxFeePerGas
+	f.dynamicPrice.TipCap = maxPriorityFeePerGas
+	return nil
+}
+
+func (f *FeeHistoryEstimator) getDynamicPrice() (fee DynamicFee, err error) {
+	f.dynamicPriceMu.RLock()
+	defer f.dynamicPriceMu.RUnlock()
+	if f.dynamicPrice.FeeCap == nil || f.dynamicPrice.TipCap == nil {
+		return fee, fmt.Errorf("dynamic price not set")
+	}
+	return f.dynamicPrice, nil
+}
+
+// BumpLegacyGas provides a bumped gas price value by bumping the previous one by BumpPercent.
+// If the original value is higher than the max price it returns an error as there is no room for bumping.
+// It aggregates the market, bumped, and max gas price to provide a correct value.
+func (f *FeeHistoryEstimator) BumpLegacyGas(ctx context.Context, originalGasPrice *assets.Wei, gasLimit uint64, maxPrice *assets.Wei, _ []EvmPriorAttempt) (*assets.Wei, uint64, error) {
+	// Sanitize original fee input
+	if originalGasPrice == nil || originalGasPrice.Cmp(maxPrice) >= 0 {
+		return nil, 0, fmt.Errorf("%w: error while retrieving original gas price: originalGasPrice: %s. Maximum price configured: %s",
+			commonfee.ErrBump, originalGasPrice, maxPrice)
+	}
+
+	currentGasPrice, err := f.getGasPrice()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	bumpedGasPrice := originalGasPrice.AddPercentage(f.config.BumpPercent)
+	bumpedGasPrice, err = LimitBumpedFee(originalGasPrice, currentGasPrice, bumpedGasPrice, maxPrice)
+	if err != nil {
+		return nil, 0, fmt.Errorf("gas price error: %s", err.Error())
```

Comment:
Could you make this error more descriptive? Something like:
```suggestion
		return nil, 0, fmt.Errorf("failed to limit bumped fee: %s", err.Error())
```

Reply:
You already know that it failed on `LimitBumpedFee` since the stack trace is going to return the `...is bumped less than minimum allowed percentage...` error. Here, we need to differentiate between the other LimitBumpedFee calls i.e. this is during gas price calculation and not maxFeePerGas.

Reply:
Still think `gas price error` doesn't seem descriptive enough. What if we say something like `failed to bump gas price`